src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataContract.cs

```diff
@@ -385,11 +326,10 @@ internal static DataContract GetDataContractSkipValidation(int id, RuntimeTypeHa
                 if (dataContract == null)
                 {
                     dataContract = CreateDataContract(id, typeHandle, type);
-                    AssignDataContractToId(dataContract, id);
                 }
```

Comment:
I'm not understanding why you removed the call to AssignDataContractToId here and in the collection data contract method.

Reply:
IIRC, this association was being done two different ways in a few different places. I decided to use one consistent method and do the association upstream.